kubernetes/kubernetes.go

```diff
@@ -217,14 +219,38 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 
 }
 
+func gatherAPIKeyFromVolume(config KubeAgentConfig, customS3Mode bool) string {
+	var key string
+	// warning message for agents that are storing the api key as an environment variable
+	if config.APIKey != "" {
+		log.Warnf("warning: cloudability api key is stored as environment variable, best practice is to" +
+			"use secret store and mount volume, see README for details")
+	}
+	if config.APIKey == "" && !customS3Mode {
```

Comment:
Would prefer flipping these conditions and attempting to load from the file first then read env var since that is the recommended approach now. Can then log separately if both are set, or fail if neither are.

I wouldn't anticipate people having both set, but if/when people switch there's certainly a potential for error/confusion that can be avoided upfront this way.